drivers/adc/ad7124/ad7124.c

```diff
@@ -761,35 +677,419 @@ int32_t ad7124_set_odr(struct ad7124_dev *dev, float odr,
+ * @brief Enable Input Buffer.
+ * @param device - AD7124 Device Descriptor.
+ * @param inbuf_en - Enable Input Buffer.
+ * @param refbuf_en - Enable reference Buffer.
+ * @param setup_id - Setup ID (Number).
+ * @return Returns 0 for success or negative error code otherwise.
+******************************************************************************/
+int ad7124_enable_buffers(struct ad7124_dev* device,
+			  bool inbuf_en,
+			  bool refbuf_en,
+			  uint8_t setup_id)
+{
+	int ret;
+	uint32_t reg_val;
+
+	if (inbuf_en)
+		/* Enable input buffer for the chosen set up. */
+		reg_val = (AD7124_CFG_REG_AIN_BUFP |
+			   AD7124_CFG_REG_AINN_BUFM);
+	else
+		reg_val =  0;
+
+	ret = ad7124_reg_write_msk(device,
+				   AD7124_CFG0_REG+setup_id,
+				   reg_val,
+				   AD7124_AIN_BUF_MSK);
+	if (ret)
+		return ret;
+
+	if (refbuf_en)
+		/* Enable reference buffer for the chosen set up */
+		reg_val = (AD7124_CFG_REG_REF_BUFP |
+			   AD7124_CFG_REG_REF_BUFM);
+	else
+		reg_val = 0;
+
+	ret = ad7124_reg_write_msk(device,
+				   AD7124_CFG0_REG+setup_id,
+				   reg_val,
+				   AD7124_REF_BUF_MSK);
+	if (ret)
+		return ret;
+
+	device->setups[setup_id].ain_buff = inbuf_en;
+	device->setups[setup_id].ref_buff = refbuf_en;
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Initializes the AD7124.
+ * @param device     - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		               parameters.
+ * @return Returns 0 for success or negative error code otherwise.
+*******************************************************************************/
+int32_t ad7124_setup(struct ad7124_dev **device,
+		     struct ad7124_init_param *init_param)
+{
+	int32_t ret;
+	struct ad7124_dev *dev;
+	uint8_t setup_index;
+	uint8_t ch_index;
+
+	dev = (struct ad7124_dev *)no_os_malloc(sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	dev->regs = init_param->regs;
+	dev->spi_rdy_poll_cnt = init_param->spi_rdy_poll_cnt;
+
+	/* Initialize the SPI communication. */
+	ret = no_os_spi_init(&dev->spi_desc, init_param->spi_init);
+	if (ret)
+		goto error_dev;
+
+	/* Update the device structure with power-on/reset settings. */
+	dev->check_ready = init_param->check_ready;
+
+	/*  Reset the device interface.*/
+	ret = ad7124_reset(dev);
+	if (ret)
+		goto error_spi;
+
+	/* Initialize ADC mode register. */
+	ret = ad7124_write_register(dev, dev->regs[AD7124_ADC_CTRL_REG]);
+	if (ret)
+		goto error_spi;
+
+	/* Get CRC State. */
+	ad7124_update_crcsetting(dev);
+	ad7124_update_dev_spi_settings(dev);
+
+	dev->active_device = init_param->active_device;
+
+	/* Read ID register to identify the part. */
+	ret = ad7124_read_register(dev, &dev->regs[AD7124_ID_REG]);
+	if (ret)
+		goto error_spi;
+	if (dev->active_device == ID_AD7124_4) {
+		if (!(dev->regs[AD7124_ID_REG].value = AD7124_4_ID))
+			goto error_spi;
+	} else if (dev->active_device == ID_AD7124_8) {
```

Comment:
This can be changed to else from else if since there seems to be only two conditions that need to be observed: ID matches AD7124_ID or not.

Reply:
Yes will change that